tests/unit/pages/Login.spec.js

```diff
@@ -96,21 +96,40 @@ describe("Login.vue", () => {
         const requestOtp = jest
           .spyOn(UserAPIService, "requestOtp")
           .mockImplementation(() => jest.fn());
-
+        const startResendOTPTimer = jest.spyOn(
+          Login.methods,
+          "startResendOTPTimer"
+        );
         mountWrapper();
+
         await setPhoneNumber();
         await requestOTP();
+        await wrapper.setData({ isResendOTPEnabled: false });
 
         // resend OTP
         await wrapper.find('[data-test="resendOTP"]').trigger("click");
 
         await flushPromises();
 
         expect(requestOtp).toHaveBeenCalled();
+        expect(startResendOTPTimer).toHaveBeenCalled();
+        expect(wrapper.vm.resendOTPTimer).toBe(60);
         expect(wrapper.vm.resentOtp).toBe(true);
         expect(wrapper.vm.invalidOtp).toBe(false);
       });
 
+      it("OTP timer", async () => {
+        //To test the Resend OTP Timer
+        //Faketimer is only needed for this test case to advance the time of interval
+        jest.useFakeTimers();
+        wrapper.vm.startResendOTPTimer(1);
```

Comment:
since your loop runs after every 1 second, best to use a value > 1 second.

Reply:
![image](https://user-images.githubusercontent.com/73693628/152785656-28c30ee4-64b0-4afc-b2be-5f2239106772.png)
Then there comes a testing error in google authentication.
And also we just need to loop it to see that after the timer is ending the button is enabling,1 works for that.